docs/source/architecture.rst

```diff
@@ -1,17 +1,142 @@
+fault tolerance, which ensures `safety and liveness
+<https://en.wikipedia.org/wiki/Liveness#Liveness_and_safety>`__.
+The network can tolerate a certain number of "bad" nodes. As long as this number
+is not exceeded, the network will work properly. In addition, blocks committed
+by nodes are final, so there are no forks in the network.
+
+All nodes on the network participate in the consensus process by verifying
+candidate blocks and exchanging consensus messages to communicate their progress
+and report on the health of the network. A `primary node` leads the network for
+a period of time, and `secondary nodes` monitor the health of the primary.
+The network switches to a new primary (called a `view change`) at regular
+intervals, as well as when the current primary is faulty.
+
+Sawtooth PBFT runs on each node in the network as a `consensus engine`, a
+separate process that handles consensus-related functionality and communicates
+with the validator through the consensus API.
+
+The following sections describe Sawtooth PBFT architecture:
+
+* :ref:`Network overview <network-overview-label>`: Describes PBFT fault
+  tolerance, view changes, sequence numbers, and the information stored
+  by each node
+
+* :ref:`Consensus messages <consensus-messages-label>`: Explains consensus
+  message structures and message types
+
+* :ref:`Sawtooth PBFT operation <pbft-operation-label>`: Shows how the
+  algorithm handles initialization, normal mode (block processing), and view
+  changes
+
+.. _network-overview-label:
+
+Network Overview
+================
+
+Fault Tolerance
+---------------
+
+A PBFT network consists of nodes that are ordered from 0 to `n-1`, where
+`n` is the total number of nodes in the network.
+
+The PBFT algorithm guarantees network `safety
+<https://en.wikipedia.org/wiki/Liveness#Liveness_and_safety>`__
+as long as the number of faulty nodes remains below the required percentage.
+The maximum number of faulty nodes that the network can tolerate is determined
+by the formula :math:`f = \frac{n - 1}{3}`. In other words, no more than one
+third of the nodes (rounded down) can be out of order or dishonest at any given
+time.
+
+For example, a four-node network (the minimum required number of nodes)
+can tolerate one faulty node. Increasing the size of the network reduces the
+likelihood that that all :math:`\frac{n - 1}{3}` nodes would be faulty at the
+same time.
+
+Network `liveness
+<https://en.wikipedia.org/wiki/Liveness#Liveness_and_safety>`__
+is guaranteed as long as there are less than :math:`2f+1` unreachable nodes,
+counting both node and network failures. That is, at least third of the nodes
+(rounded up) must be reachable at all times.
+
+View Changes: Choosing a New Primary
+------------------------------------
+
+A PBFT network has one leader, called the `primary node`. All other nodes on the
+network are called `secondary nodes`.
+
+* The primary requests new blocks from the validator, notifies the network when
```

Comment:
I would simplify the requests new blocks/notifies the network bit to just saying that the primary builds and publishes new blocks.

Reply:
I was trying to avoid saying that the primary node publishes the new blocks.  My understanding is that the primary tells the validator that it's okay for the validator to publish the block.

Reply:
Or does the consensus engine actually handle the mechanics of publishing a block?

Reply:
After chatting, I agree that we don't need to make this distinction at the beginning of the section. (It's described in more detail later.)